pkg/operator/ceph/object/objectstore.go

```diff
@@ -525,18 +530,30 @@ func createSystemUser(objContext *Context, namespace string) error {
 	return nil
 }
 
-func setMultisite(objContext *Context, store *cephv1.CephObjectStore, serviceIP string) error {
+func setMultisite(objContext *Context, store *cephv1.CephObjectStore) error {
 	logger.Debugf("setting multisite configuration for object-store %v", store.Name)
-	serviceEndpoint := fmt.Sprintf("http://%s:%d", serviceIP, store.Spec.Gateway.Port)
-	if store.Spec.Gateway.SecurePort != 0 {
-		serviceEndpoint = fmt.Sprintf("https://%s:%d", serviceIP, store.Spec.Gateway.SecurePort)
-	}
+	serviceEndpoint := objContext.Endpoint
 
 	if store.Spec.IsMultisite() {
+		if store.Spec.Zone.Endpoint != "" {
+			serviceEndpoint = store.Spec.Zone.Endpoint
```

Comment:
I'm also confused. What's happening here? I don't see anywhere where Rook is setting the default value for this if it isn't set. Obviously, Rook must be doing this, but I can't figure out how, and our code shouldn't be a mystery. We should strive to make it as clear as we can.

Reply:
What this conditional block is doing is saying if the object-store spec has a `zone` section and if that `zone` section has an `endpoint` then set the serviceEndpoint that defined Endpoint.

The check `if store.Spec.Zone.Endpoint != ""` is same kind of check the `IsMultisite()` function the line above is using (https://github.com/rook/rook/blob/master/pkg/apis/ceph.rook.io/v1/object.go#L31). If a section or config variable isn't defined in the .yaml file for the CRD then it is `== ""`. Is there some better way to check for whether or not a section or CRD option is defined?

Also I'm not sure what Travis means by setting the endpoint in the cluster CR or why the default value for the endpoint is a concern but I hope my comment above addressed those two concerns.

Reply:
The `endpoint` of obj context is filled using the rgw-service DNS name. The zone has list of endpoints of different rgw servers. By default we set it with rgw-service DNS name, but if ObjectStore.Spec.Zone.Endpoint is defined then that value is used.

Reply:
Ok so this is set by default based on the dns name in the context: `serviceEndpoint := objContext.Endpoint`.
And if it's multisite, it can be overridden by the new endpoint setting on the zone. 
So if they don't override the setting in the zone, the default might work? Or in the multisite scenario is the endpoint required?

Reply:
> So if they don't override the setting in the zone, the default might work? Or in the multisite scenario is the endpoint required?

The default is set as an IP, rather than the svc name, the operator gets the svc IP.  See here: https://github.com/rook/rook/blob/3005a6fc6846992ff14a14a910e1f4b5066a52ad/pkg/operator/ceph/object/objectstore.go#L530

So, unless both you are operating 2 ceph zones on the same k8s cluster, or those svc IPs are routable between two separate k8s clusters, the default is not useful for most real-world scenarios.  Most users will want to explicitly set the the endpoint value.

Reply:
@mgugino-uipath I have also found recently that (at least in the default minikube setup) kubernetes services are not routable to other services. I have had trouble even getting 2 zones communicating inside the same k8s cluster. See https://github.com/rook/rook/issues/10489

Reply:
Also, I see now that I missed where objContext.Endpoint was being set with `BuildDNSEndpoint()`. I think this seems good now knowing that.

Reply:
Earlier the `endpoint` for zone/zonegroup was set as `Service.IP` but while removing zone/zonegroup they were trying to remove objContext.Endpoint. I fixed that in the one commit in this PR make it consistent in both cases. Now with the new change if the endpoint is defined in `ObjectStore.SPec.Zone.Endpoint` then will that value otherwise be the value from `objectContext.Endpoint`